### PR TITLE
make it possible to run pipeline on arm64

### PR DESCRIPTION
**Pull request by 6543**
test pipeline to run on arm64

```
Architecture:           aarch64
  CPU op-mode(s):       32-bit, 64-bit
  Byte Order:           Little Endian
CPU(s):                 8
  On-line CPU(s) list:  0-7
Vendor ID:              ARM
  Model name:           Neoverse-N1
    Model:              1
    Thread(s) per core: 1
    Core(s) per socket: 8
    Socket(s):          1
    Stepping:           r3p1
    BogoMIPS:           50.00
    Flags:              fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs
```

## TODOs

- [x] web: [works](https://ci.woodpecker-ci.org/repos/3780/pipeline/7363/46)
- [ ] test
  - [x] lint-editorconfig: ~~[fail](https://ci.woodpecker-ci.org/repos/3780/pipeline/7361/27)~~ -> works after image was updated on dockerhub
  - [x] codecov: fail ... will be removed
  - [ ] mysql: [failed](https://ci.woodpecker-ci.org/repos/3780/pipeline/7365/42) -> service-mysql does not have arm64 image
- [x] securityscan:  [works](https://ci.woodpecker-ci.org/repos/3780/pipeline/7363/46)
- [x] docs: [works](https://ci.woodpecker-ci.org/repos/3780/pipeline/7695/28)
- [ ] docker
  - [ ] cross-compile-server: [fail](https://ci.woodpecker-ci.org/repos/3780/pipeline/7366/20) -> https://github.com/techknowlogick/xgo/issues/213
  - [x] build-agent/cli: [fail](https://ci.woodpecker-ci.org/repos/3780/pipeline/7366/11) -> unrelated -> fixed by #2240
- [ ] binaries
  - [ ] cross-compile-server: [fail](https://ci.woodpecker-ci.org/repos/3780/pipeline/7366/10) -> https://github.com/techknowlogick/xgo/issues/213

### CODE REVIEW

**Comment by 6543** [comment]
PS: we need better error message than `exec XXXXXX: exec format error` on our docker backend side ... to show that it's an cpu architecture mismatch between image and agent

**Comment by anbraten** [comment]
> PS: we need better error message than `exec XXXXXX: exec format error` on our docker backend side ... to show that it's an cpu architecture mismatch between image and agent 

That's coming from docker

File: .woodpecker/test.yml [hunk 1]
```diff
@@ -61,9 +61,10 @@ steps:
       - "[ -n \"$DIFF\" ] && { echo \"swagger not up to date, exec 'make generate-swagger' and commit\"; exit 1; } || true"
     when: *when
 
-  lint-editorconfig:
-    image: mstruebing/editorconfig-checker
-    group: test
+# TODO: fail on arm
+# lint-editorconfig:
+#   image: mstruebing/editorconfig-checker
+#   group: test
```
Comment (6543) [inline .woodpecker/test.yml:67]:
```suggestion
 lint-editorconfig:
   image: codeberg.org/6543/docker-images/editorconfig-checker
   group: test
```

File: .woodpecker/test.yml [hunk 2]
```diff
@@ -61,9 +61,9 @@ steps:
       - "[ -n \"$DIFF\" ] && { echo \"swagger not up to date, exec 'make generate-swagger' and commit\"; exit 1; } || true"
     when: *when
 
-  lint-editorconfig:
-    image: mstruebing/editorconfig-checker
-    group: test
+ lint-editorconfig:
+   image: codeberg.org/6543/docker-images/editorconfig-checker
+   group: test
```
Comment (6543) [inline .woodpecker/test.yml:66]:
```suggestion
  lint-editorconfig:
    image: codeberg.org/6543/docker-images/editorconfig-checker
    group: test
```

**Comment by pat-s** [comment]
@6543 I updated the OP with the current state. Essentially only `xgo` is missing and the question whether to bump to mysql v8 as v5 will likely never get an arm image.

**Comment by 6543** [comment]
we can bump the mysql version ... would you make a pull for it?

**Comment by pat-s** [comment]
@qwerty287 The PR in upstream xgo got reverted due to some issues. Hence, there is still no arm64 xgo image and subsequently this needs to stay open.

**Comment by 6543** [comment]
->  #3955

**Pull request closed without merging**